File: client-clojure/src/sirqul_io_t_platform/specs/purchase_order_item_response.clj
(ns sirqul-io-t-platform.specs.purchase-order-item-response
  (:require [clojure.spec.alpha :as s]
            [spec-tools.data-spec :as ds]
            [sirqul-io-t-platform.specs.name-string-value-response :refer :all]
            [sirqul-io-t-platform.specs.account-short-response :refer :all]
            )
  (:import (java.io File)))


(def purchase-order-item-response-data
  {
   (ds/opt :valid) boolean?
   (ds/opt :message) string?
   (ds/opt :version) float?
   (ds/opt :serializeNulls) boolean?
   (ds/opt :startTimeLog) int?
   (ds/opt :errorCode) string?
   (ds/opt :request) (s/coll-of name-string-value-response-spec)
   (ds/opt :purchaseOrderItemId) int?
   (ds/opt :created) int?
   (ds/opt :updated) int?
   (ds/opt :customer) account-short-response-spec
   (ds/opt :orderItemType) string?
   (ds/opt :orderItemId) int?
   (ds/opt :orderCustomType) string?
   (ds/opt :orderCustomId) string?
   (ds/opt :retailerLocationId) int?
   (ds/opt :amount) float?
   (ds/opt :tax) float?
   (ds/opt :comment) string?
   (ds/opt :returning) string?
   })

(def purchase-order-item-response-spec
  (ds/spec
    {:name ::purchase-order-item-response
     :spec purchase-order-item-response-data}))

File: client-scalaz/src/main/scala/org/openapitools/client/api/PurchaseOrderItemResponse.scala
package org.openapitools.client.api

import argonaut._
import argonaut.EncodeJson._
import argonaut.DecodeJson._

import org.http4s.{EntityDecoder, EntityEncoder}
import org.http4s.argonaut._
import org.joda.time.DateTime


import PurchaseOrderItemResponse._

case class PurchaseOrderItemResponse (
  valid: Option[Boolean],
message: Option[String],
version: Option[Double],
serializeNulls: Option[Boolean],
startTimeLog: Option[Long],
errorCode: Option[String],
request: Option[List[NameStringValueResponse]],
purchaseOrderItemId: Option[Long],
created: Option[Long],
updated: Option[Long],
customer: Option[AccountShortResponse],
orderItemType: Option[String],
orderItemId: Option[Long],
orderCustomType: Option[String],
orderCustomId: Option[String],
retailerLocationId: Option[Long],
amount: Option[Double],
tax: Option[Double],
comment: Option[String],
returning: Option[String])

object PurchaseOrderItemResponse {
  import DateTimeCodecs._

  implicit val PurchaseOrderItemResponseCodecJson: CodecJson[PurchaseOrderItemResponse] = CodecJson.derive[PurchaseOrderItemResponse]
  implicit val PurchaseOrderItemResponseDecoder: EntityDecoder[PurchaseOrderItemResponse] = jsonOf[PurchaseOrderItemResponse]
  implicit val PurchaseOrderItemResponseEncoder: EntityEncoder[PurchaseOrderItemResponse] = jsonEncoderOf[PurchaseOrderItemResponse]
}
